File: insurance-management/src/app/interfaces/policy.ts
import { User } from './user';
import { PolicyType } from './policy-type';
import { Beneficiary } from './beneficiary';
import { Company } from './company';
export interface Policy {
  uuid: string; // This will be a UUID string
  owner: User; // Reference to the User (owner of the policy)
  company?: Company; // Reference to the Company that owns this policy
  policyNumber: string;
  insuranceCompany: string;
  lastUpdated: string; // ISO 8601 string or Date string
  createdAt: string; // ISO 8601 string or Date string
  policyType: PolicyType; // Enum or string representing policy type


  coverageAmount: number;
  monthlyPremium: number;
  status: string;
  beneficiaries: Beneficiary[]; // List of Beneficiaries

  addressLine1: string;
  city: string;
  region: string;
  postalCode: string;
  country: string;
}
/*

export interface AddPolicy {
  policyNumber: string;
  insuranceCompany: string;
  policyType: string;

  beneficiaries: Beneficiary[];
}
  */

File: insurance-backend/src/main/java/co/za/insurance/policy/Policy.java
package co.za.insurance.policy;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import com.earmitage.core.security.repository.User;
import co.za.insurance.company.Company;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Getter
@Setter
public class Policy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private final String uuid = UUID.randomUUID().toString();

    @ManyToOne
    private User owner;

    @ManyToOne
    private Company company;

    private String policyNumber;
    private String insuranceCompany;
    private LocalDateTime lastUpdated;
    private LocalDateTime createdAt;
    private PolicyType policyType;
    
    private Double coverageAmount;
    private Double monthlyPremium;
    private String status;
    
    private String addressLine1;
    private String city;
    private String region;
    private String postalCode;
    private String country;
    
    @OneToMany(mappedBy = "policy", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Beneficiary> beneficiaries;
    
    private Boolean deceased;
    private LocalDate deceasedDate;

}